Updating the storage-users service description (#745)
owncloud/docs-ocis: modules/ROOT/pages/deployment/services/s-list/storage-users.adoc

[source,plaintext]
----
COMMANDS:
   sessions  Print a list of upload sessions
   clean     Clean up leftovers from expired uploads
----

=== Manage Trash-Bin Items

> This command set is about the trash-bin to get an overview of items, restore items and purging old items of `personal` spaces and `project` spaces (spaces that have been created manually). `trash-bin` commands require a `spaceID` as parameter. See the xref:maintenance/space-ids/space-ids.adoc[Listing Space IDs] for details how to get them.

Reviewer: ```suggestion
This command set provides commands to get an overview of trash-bin items, restore items and purge old items of `personal` spaces and `project` spaces (spaces that have been created manually). `trash-bin` commands require a `spaceID` as parameter. See xref:maintenance/space-ids/space-ids.adoc[Listing Space IDs] for details of how to get them.
```

[source,bash]
----
ocis storage-users trash-bin <command>
----

[source,plaintext]
----
COMMANDS:
   purge-expired  Purge expired trash-bin items
   list           Print a list of all trash-bin items of a space.
   restore-all    Restore all trash-bin items for a space.
   restore        Restore a trash-bin item by ID.
----

==== Purge Expired

Purge all expired items from the trash-bin.

[source,plaintext]
----
ocis storage-users trash-bin purge-expired
----

The behaviour of the `purge-expired` command can be configured by using the following environment variables.

* `STORAGE_USERS_PURGE_TRASH_BIN_USER_ID` +
Used to obtain space trash-bin information and takes the system admin user as the default which is the `OCIS_ADMIN_USER_ID` but can be set individually. It should be noted, that the `OCIS_ADMIN_USER_ID` is only assigned automatically when using the single binary deployment and must be manually assigned in all other deployments. The command only considers spaces to which the assigned user has access and delete permission.

* `STORAGE_USERS_PURGE_TRASH_BIN_PERSONAL_DELETE_BEFORE` +

> Has a default value of `30 days`, which means the command will delete all files older than `30 days`. The value is human-readable, valid values are `24h`, `60m`, `60s` etc. `0` is equivalent to disable and prevents the deletion of `personal space` trash-bin files.

Reviewer: To specify "14 days", can I write `14d` (similar to `24h` `60m`)
If so, then it would be useful to add that as an example of valid values.

Author: You highlighted an interersting issue that has slipped thru my attention...
I needed to rephrased the sentence because the duration type is limited to values of h and below but not days. We have a page describing value types which I also linked.

> Has a default value of `30 days`, which means the command will delete all files older than `30 days`. The value is human-readable, valid values are `24h`, `60m`, `60s` etc. `0` is equivalent to disable and prevents the deletion of `personal space` trash-bin files.

Author: ```suggestion
Has a default value of `720h` which equals `30 days`. This means, the command will delete all files older than `30 days`. The value is human-readable, for valid values see the duration type described in the xref:deployment/services/envvar-types-description.adoc[Environment Variable Types]. A value of `0` is equivalent to disable and prevents the deletion of `personal space` trash-bin files.
```
After clarification, we have to stick to teh duration type. I rephrased the sentence accordingly and added a link to the corresponding description.

* `STORAGE_USERS_PURGE_TRASH_BIN_PROJECT_DELETE_BEFORE` +
Has a default value of `30 days`, which means the command will delete all files older than `30 days`. The value is human-readable, valid values are `24h`, `60m`, `60s` etc. `0` is equivalent to disable and prevents the deletion of `project space` trash-bin files.

==== List and Restore Trash-Bins Items

> The variable `STORAGE_USERS_CLI_MAX_ATTEMPTS_RENAME_FILE` defines a maximum number of attempts to rename a file when the admin restores the file with the CLI option `--option keep-both` to existing destination with the same name.

Reviewer: ```suggestion
The variable `STORAGE_USERS_CLI_MAX_ATTEMPTS_RENAME_FILE` defines a maximum number of attempts to rename a file when the admin restores the file with the CLI option `--option keep-both` to an existing destination with the same name.
```

Is it possible to provide some reasons why someone might want to set this, what might be a "reasonable" value, and what might be the next steps if a CLI user has this failing after the retries.

{empty}

Print a list of all trash-bin items of a space::
+
--
[source,bash]
----
ocis storage-users trash-bin list
----
--

Restore all trash-bin items for a space::
+
--
[source,bash]
----
ocis storage-users trash-bin restore-all
----
--

Restore a trash-bin item by ID::
+
--
[source,bash]
----

> ocis storage-users trash-bin restore

Reviewer: I suppose this needs an id (of some sort) as a parameter?
